che-terminal-client/pom.xml

```diff
@@ -57,42 +57,45 @@
                 </configuration>
             </plugin>
             <plugin>
-                <artifactId>maven-antrun-plugin</artifactId>
+                <groupId>com.github.eirslett</groupId>
+                <artifactId>frontend-maven-plugin</artifactId>
                 <executions>
                     <execution>
-                        <!--clean-up build folders-->
-                        <id>clean-up</id>
-                        <phase>clean</phase>
+                        <id>install node and npm</id>
+                        <phase>generate-resources</phase>
                         <goals>
-                            <goal>run</goal>
+                            <goal>install-node-and-npm</goal>
                         </goals>
                         <configuration>
-                            <tasks>
-                                <delete dir="${basedir}/node_modules" />
-                                <delete dir="${basedir}/build" />
-                            </tasks>
+                            <arguments>install --fetch-retry-mintimeout 60000 --fetch-retries 10 --maxsockets 2</arguments>
                         </configuration>
                     </execution>
                     <execution>
-                        <!--compilation sources-->
-                        <id>compilation</id>
+                        <id>npm install</id>
                         <phase>compile</phase>
                         <goals>
-                            <goal>run</goal>
+                            <goal>npm</goal>
                         </goals>
                         <configuration>
-                            <target>
-                                Download NPM dependencies
-                                <exec dir="${basedir}" executable="npm" failonerror="true">
-                                    <arg value="install" />
-                                </exec>
-                                <exec dir="${basedir}" executable="npm" failonerror="true">
-                                    <arg value="run-script" />
-                                    <arg value="build" />
-                                </exec>
-                            </target>
+                            <arguments>install browserify vinyl-buffer fs-extra gulp merge-stream sorcery vinyl-source-stream gulp-sourcemaps gulp-typescript gulp-uglify typescript --save-dev --fetch-retry-mintimeout 60000 --fetch-retries 10 --maxsockets 2</arguments>
```

Comment:
AFAIK we shouldn't do this, all packages that are required are defined in package.json so doing that is kind of bypassing the file and the versions